media/js/base/core-datalayer.js

```diff
@@ -0,0 +1,34 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// create namespace
+if (typeof Mozilla == 'undefined') {
+    var Mozilla = {};
+}
+
+/**
+* Utility class for core dataLayer object to track contextual user and page data
+*/
+
+(function() {
+    var dataLayer = window.dataLayer = window.dataLayer || [];
+    var Analytics = {};
+
+    // Reurns page ID used in Event Category for GA events tracked on page
+    Analytics.getPageId = function() {
+        var pageId = $('html').attr('data-gtm-page-id');
+
+        return pageId ? pageId : document.location.pathname.replace(/\/[^\/]*/, '');
+    };
+
+    // Push core datalayer object
+    dataLayer.push({
+        'event': 'core-dataLayer-loaded',
+        'pageId': Analytics.getPageId()
```

Comment:
Your request here makes sense, but I don't fully understand the benefit of having the test run before the functions are called? Am curious for my own benefit and so I make better decisions next time.